File: tree.cpp
#include <iostream>

using namespace std;

class node
{
    public:
    int val;
    node *left;
    node *right;

    node(int v)
    {
        this->val=v;
        this->left=NULL;
        this->right=NULL;
    }
};

class bintree
{
    public:
    node *root;

    bintree()
    {
        this->root=NULL;
    }

    void add(int v)
    {
        node *newnode=new node(v);
        if(this->root==NULL)
        {
            this->root=newnode;
            return;
        }
        node *temp=root;
        while(true)
        {
            if(v<(temp->val))
            {
                if(temp->left!=NULL)
                {
                    temp=temp->left;
                }
                else{
                    temp->left=newnode;
                    break;
                }
            }
            else{
                if(temp->right!=NULL)
                {
                    temp=temp->right;
                }
                else{
                    temp->right=newnode;
                    break;
                }
            }
        }
    }

};

void inorder(node *root)
{
    if(root!=NULL)
    {
        inorder(root->left);
        cout<<root->val<<" ";
        inorder(root->right);
    }
}
int main()
{
    bintree b;
    b.add(5);
    b.add(3);
    b.add(7);
    b.add(2);
    b.add(4);
    b.add(6);
    b.add(8);

    inorder(b.root);


}

File: tree.py
class node:
    def __init__(self,n):
        self.val=n
        self.left=None
        self.right=None

class Trees:
    def __init__(self):
        self.root=None
    
    def add(self,n):

        if(self.root==None):
            self.root=node(n)
        
        else:
            temp=self.root
            while True:
                if(n<temp.val):
                    if temp.left==None:
                        temp.left=node(n)
                        break;
                    else:
                        temp=temp.left
                else:
                    if(temp.right==None):
                        temp.right=node(n)
                        break
                    else:
                        temp=temp.right


def inorder(root):
    if(root!=None):
        inorder(root.left)
        print(root.val,end=" ")
        inorder(root.right)
                
                
t=Trees()
t.add(5)
t.add(3)
t.add(8)

inorder(t.root)